src/components/shared/Modal.tsx

```diff
@@ -20,26 +20,56 @@ export type InputFields = {
   address: string;
 };
 
-export function Modal({ closeModal, addPatient }: ModalProps) {
+export function Modal({ closeModal, addPatient, currentPatient }: ModalProps) {
   const {
     register,
     handleSubmit,
+    setValue,
+    reset,
     formState: { errors },
   } = useForm<InputFields>({
     // resolver: yupResolver(RegistrationSchema),
   });
 
+  const { createPatient, editPatient } = usePatientsActions();
+
+  useEffect(() => {
+    if (currentPatient) {
+      const [firstName = "", lastName = ""] = currentPatient.name.split(" ");
+      setValue("firstName", firstName);
+      setValue("lastName", lastName);
+      setValue("idNumber", currentPatient.id || "");
+      setValue("dateBirth", currentPatient.dateBirth || "");
+      setValue("mobile", currentPatient.mobile || "");
+      setValue("email", currentPatient.email || "");
+      setValue("address", currentPatient.address || "");
+    } else {
+      reset();
+      reset({
+        firstName: "",
+        lastName: "",
+        idNumber: "",
+        dateBirth: "",
+        mobile: "",
+        email: "",
+        address: "",
+      });
+    }
+  }, [currentPatient, setValue, reset]);
+
   const onSubmit: SubmitHandler<InputFields> = async (data) => {
     const newPatient: User = {
       name: `${data.firstName} ${data.lastName}`,
       ...data,
-      id: "",
+      id: currentPatient?.id || "",
     };
 
     try {
-      const patientsCollectionRef = collection(db, "patients");
-      await addDoc(patientsCollectionRef, newPatient);
-
+      if (currentPatient) {
+        await editPatient(newPatient);
+      } else {
+        await createPatient(newPatient);
+      }
```

Comment:
Ja bym to całe rozbił, i zrobił w onSubmicie ifa isEditMode (mozesz to po prostu sprawdzic czy jest currentPatient co swoja droga jest slaba nazwa, wiec lepiej to zrobic na isEditMode)

Jezeli jest editMode to robimy sobie body ktory chcemy wyslac i puszczamy editPatient
Jezeli nie jest to robimy sobie inne body ktore chcemy wyslac i puszczamy createPatient

Bedzie czytelniej myślę